Explain saucectl architecture (#1520)
saucelabs/sauce-docs: docs/dev/cli/saucectl/architecture.md

---
id: architecture
title: saucectl Architecture
sidebar_label: Architecture
---

This document provides an overview of the `saucectl` CLI architecture to help you better understand how it works.

## Architecture Components

The following provides a high-level overview of the `saucectl` CLI architecture.

1. `saucectl` sends the test payload (mobile app, test app, or project files) to the app storage.
2. Depending on the capabilities of the underlying test framework (e.g., Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.
3. The device cloud allocates a device for your request, retrieves the payload from app storage, and runs the tests.

This diagram illustrates at high level how `saucectl` interacts with the Sauce Labs environment.

<img src="/img/saucectl/components.jpg" />

> More Information

Reviewer: ```suggestion
## More Information
```

> * [Using the saucectl CLI](/dev/cli/saucectl).

Reviewer: ```suggestion
* [Using the saucectl CLI](/dev/cli/saucectl)
```

---
id: architecture
title: saucectl Architecture
sidebar_label: Architecture
---

This document provides an overview of the `saucectl` CLI architecture to help you better understand how it works.

> ## What You'll Need

Reviewer: Is this heading needed? The reference to 'Using the saucectl CLI' could be moved to the bottom of the page as a link "for more information".

Reviewer 2: Moved to end of page.

If you're new to `saucectl`, we recommend reviewing the [Using the saucectl CLI](/dev/cli/saucectl) documentation.

> ## Architecture

Reviewer: The H2 heading for "Architecture" seems a bit redundant/repetitive as it's included in the title/H1 heading.
```suggestion
## Architecture Components
```

Here's what happens in a nutshell:

1. `saucectl` sends the test payload (mobile app, test app, or project files) to the app storage.

> 2. Depending on the capabilities of the underlying test framework (say, Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.

Reviewer: ```suggestion
2. Depending on the capabilities of the underlying test framework (e.g., Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.
```
